src/workerd/io/io-context.c++

```diff
@@ -1234,7 +1242,7 @@ jsg::Promise<kj::Maybe<IoOwn<kj::AsyncInputStream>>> IoContext::makeCachePutStre
     // This Cache API subrequest would exceed the total PUT quota. There used to be a limit on
     // individual puts, but now this is only used as a fast path for rejecting PUTs going beyond the
     // total quota early.
-    if (length > MAX_TOTAL_PUT_SIZE) {
+    if (length > initialPutQuota) {
```

Comment:
That would be possible, but with that approach every class inheriting from LimitEnforcer would have to define a quota instead of being able to use kj::None for the default. By having the default defined in here and initialPutQuota specifying the actual initial quota, we only need to define the default once. We don't have a clear approach to how limits are being handled in workerd, so instead of trying to handle that thorny question now I sought to keep this close to the existing implementation.

Another approach to handle this would be to define the default limit in a non-virtual function in LimitEnforcer, but I assumed that since the class is entirely virtual so far this may have some side effects.

Reply:
Perhaps leave a TODO comment to encourage us to revisit this later to see if there's a reasonable way to move this into `limitEnforcer`?

Reply:
I'm not familiar with this work, but if implementing a stateful limit for a request, it might be worth noting that all active requests for a DO share an IoContext.  To allow DO requests to effectively share limits, we have a `topUpActor()` method in `LimitEnforcer` that is called to reset the limits each time a new request arrives.

Reply:
👍  Added a TODO for `initialPutQuota` in `src/workerd/io/io-context.h`.